modulus/experimental/resdiff/generate.py

```diff
@@ -213,10 +214,10 @@ def randint(self, *args, size, **kwargs):
         return torch.stack([torch.randint(*args, size=size[1:], generator=gen, **kwargs) for gen in self.generators])
 
 
-def load_pickle(network_pkl):
+def load_pickle(network_pkl, dist):
```

Comment:
something to consider is instead of making this a parameters, just create dist in side the method everytime. Up to you, but the singleton allows you to keep the API clean.

Reply:
Another suggestion is to lift the check up (out of `load_pickle`) and have something like:
```py
if dist.rank == 0:
    load_pickle(pkl1)
    load_pickle(pkl2)
   ...
```

Reply:
Thanks for the comment. I'm now instantiating the `dist` object inside the method